### ISSUE TITLE
 create_codemeta()  may lead to 'Error in !auto : invalid argument type'

### ISSUE THREAD

**Issue by RLumSK** [ropensci/codemetar#238]
First thanks for this package, I think it is of great value for the community!

I noticed that `create_codemeta()` (path is the package path) stops with throwing

```
Error in !auto : invalid argument type 
```

This error is caused by the function `read_citation_with_encoding()` (parse_citation.R) which calls `utils::readCitationFile(citation_file, meta = meta)` and it happens if the `CITATION` file contains the following line: 

```r
citation(auto = meta)
```

This is nothing special and works on CRAN, since `meta` comes with the correct information the moment the package is built on CRAN. I raised a similar issue a while ago in 'pkgdown' 
(https://github.com/r-lib/pkgdown/issues/493), for which a working solution was found.

**Comment by cboettig** [comment]
Thanks for the detailed report!  Looks like we could probably bootstrap something based on the `pkgdown` solution... @maelle any thoughts?

**Comment by maelle** [comment]
Thanks for the bug report! @cboettig I am on maternity leave so can't look into code at the moment. 👶

**Comment by RLumSK** [comment]
I proposed a solution to this problem with my PR, this solution is a little bit different from what was done in 'pkgdown', however, it addresses the issue and the function does not break anymore.

**Comment by cboettig** [comment]
closed by #241
